Backend/IDatabase.cs

```diff
@@ -4,37 +4,46 @@ namespace Backend
 {
     public interface IDatabase
     {
-        public int GetLastCarId();
+        //public int GetLastCarId();
 
         // returns: <amount> of vehicles from beginning of the sorted list serialized to UTF-8 JSON
-        public byte[] GetSortedCarsJson(SortingCriteria sortBy, bool sortAscending, int startIndex, int amount);
+        public IEnumerable<Car> GetSortedCars(SortingCriteria sortBy, bool sortAscending, int startIndex, int amount);
 
         // returns: <amount> of vehicles from beginning of the filtered and sorted list serialized to UTF-8 JSON
-        public byte[] GetFilteredCarsJson(CarFilters filters, SortingCriteria sortBy, bool sortAscending, int startIndex, int amount);
-
+        public IEnumerable<Car> GetFilteredCars(CarFilters filters, SortingCriteria sortBy, bool sortAscending, int startIndex, int amount);
 
+        public Car GetCar(int id);
         // args: car serialized to UTF-8 JSON
-        public bool AddCarJson(byte[] car);
+        public bool AddCar(string username, string password, Car car);
 
+        public bool UpdateCar(string username, string password, Car car);
         // returns: true, if removal is successful, false if not
-        public bool DeleteCar(int id);
-
+        public bool DeleteCar(int carId, string username, string password);
```

Comment:
Password should never appear as a string in code. Check the user token instead. https://en.wikipedia.org/wiki/Access_token

Reply:
https://en.wikipedia.org/wiki/Representational_state_transfer
Quoting from there: "...using a uniform and predefined set of stateless operations"
At least as far as I know REST APIs should not store temporary state (such as access tokens).
Unless we are not making REST API

Reply:
![image](https://user-images.githubusercontent.com/26715393/100443400-4ed7a780-30b2-11eb-8cb4-76b1a02212fc.png)

https://www.oauth.com/oauth2-servers/access-tokens/#:~:text=Access%20tokens%20are%20the%20thing,in%20transit%20and%20in%20storage.

Reply:
Our current implementation is keeping user credentials private.
At least I see no big problem with current implementation.

Reply:
Will use HTTP Basic authentication (at least for now)